### ISSUE TITLE
[8] Gradle support in mercator

### ISSUE THREAD

**Issue by msrb** [openshiftio/openshift.io#689]
# User story

As a fabric8-analytics developer, I want to be able to extract dependencies from [Gradle](https://gradle.org/) manifest files. This will enable me to collect dependency usage information from Java projects which use Gradle as a build system. 

# Acceptance criteria
- mercator supports Gradle
- mercator doesn't execute any untrusted code from Gradle manifest files
- mercator with Gradle support is available in production

# Tasks
- [x] implement support for Gradle manifest files in mercator
- [x] build an RPM package
- [x] update base worker image to pull in the latest mercator RPM
- [x] update worker's base image
- [x] push the changes to production

**Comment by jpopelka** [comment]
@pkajaba haven't we had some document from a spike around Gradle ?

**Comment by pkajaba** [comment]
https://drive.google.com/open?id=1TLze74pitjyN5iAZAxijZcLgS7s-zHbL-SYX0GcUCK4 @jpopelka

**Comment by msrb** [comment]
Since we rely on GitHub to identify most used components, we need to add support for Gradle. Gradle is becoming more and more popular: https://github.com/search?utf8=%E2%9C%93&q=filename%3Abuild.gradle&type=Code

**Comment by krishnapaparaju** [comment]
@msrb Are we working on this initiative ?

**Comment by msrb** [comment]
@krishnapaparaju not in this sprint, but I will prioritize it in the next.

**Comment by pkajaba** [comment]
@krishnapaparaju I have concluded that only safe option how to gather data from `Gradle` files is simply parse those files. gradle offers a good tooling, where we could just print dependencies with `gradle dependencies` and much more stuff! However, Gradle files are scripts. Users can put there whatever they want and it will be executed. This might get us into the trouble (credentials leak, database corruption, etc.)

That's why I decided to stick with static analyzing. This approach has one big disadvantage, because if package's dependencies are resolved during the run of `Gradle` dependencies. We will get a name of the variable, where the version of dependency should be stored for example.
